### PR TITLE
UpdateTestcase

### PR DESCRIPTION
**Pull request by pranjaltale16**
Added some more base test cases. 
Updated `test.sh` such that, one can just append the name of the folder in the `dirList` and it will run tests for all the registered directories.

### CODE REVIEW

File: test.sh [hunk 1]
```diff
@@ -16,7 +16,10 @@
 
 source base.sh
 
-#Test for codeworld-compiler
-run codeworld-compiler cabal configure --enable-tests
-run codeworld-compiler cabal build 
-run codeworld-compiler cabal test
+DirList="codeworld-compiler"
```
Comment (cdsmith) [inline test.sh]: If you wanted to be even fancier, you could try:

    for i in $(grep -l -i Test-Suite */*.cabal); do
        TARGET=$(basename $i)
        run $TARGET cabal configure --enable-tests
        run $TARGET cabal build
        run $TARGET cabal test
    done

What do you think?

Comment (pranjaltale16) [inline test.sh]: The problem with base case is, it is resulting in `directory-name.cabal` while run function needs the name of the directory, so I updated it with `cut`. Is it okay?

Comment (cdsmith) [inline test.sh]: Oops, my mistake!  I meant `dirname`.  Your version works, but `dirname` is probably clearer.

Comment (pranjaltale16) [inline test.sh]: Done, please have a look.

**Comment by cdsmith** [comment]
Thanks!

**Pull request merged**